src/auth0-sso-login.js

```diff
@@ -38,32 +47,64 @@ export default class auth {
     });
   }
 
-  // calls a hook once the profile got refreshed
+  /**
+   * @description the latest authorization result with access token
+   * @return {null|Object} authResult if the user was already logged in; null otherwise
+   */
+  getLatestAuthResult() {
+    return this.authResult;
+  }
+
+  /**
+   * @description calls a hook once the profile got refreshed
+   * @param profile user profile retrieved from auth0 manager
+   * @return {*}
+   */
   profileRefreshed(profile) {
     if (this.config.hooks && this.config.hooks.profileRefreshed) {
       return this.config.hooks.profileRefreshed(profile);
     }
     return Promise.resolve();
   }
 
-  // calls a hook once the token got refreshed
+  /**
+   * @description Calls a hook once the token got refreshed
+   * @param authResult authorization result returned by auth0
+   * @return {*}
+   */
   tokenRefreshed(authResult) {
+    this.authResult = authResult;
+    this.tokenExpiryManager.scheduleTokenRefresh(authResult,
+      () => this.ensureLoggedIn({ enableLockWidget: true, forceTokenRefresh: true }));
+
     if (this.config.hooks && this.config.hooks.tokenRefreshed) {
       return this.config.hooks.tokenRefreshed(authResult);
     }
     return Promise.resolve();
   }
 
-  // calls a hook once the login should be removed
+  /**
+   * Calls a hook once the login should be removed
+   * @return {*}
+   */
   removeLogin() {
+    this.tokenExpiryManager.cancelTokenRefresh();
+    this.authResult = null;
+
     if (this.config.hooks && this.config.hooks.removeLogin) {
       return this.config.hooks.removeLogin();
     }
     return Promise.resolve();
   }
 
-  // calls a hook to log out the user, and then interacts with Auth0 to actually log the user out
+  /**
+   * @description Calls a hook to log out the user, and then interacts with Auth0 to actually
+   * log the user out.
+   */
   logout() {
+    this.tokenExpiryManager.cancelTokenRefresh();
+    this.authResult = null;
+
     if (this.config) {
       if (this.config.hooks && this.config.hooks.logout) {
```

Comment:
Should we call the `removeLogin` hooks too?  We might want to look at a client implementation to see if it makes sense to call that.

Reply:
For a client, these actions probably mean the same, as the result is the same => no access token available. I can imagine that some clients can have some additional logic for `logout` compared to `removeLogin`. Therefore `removeLogin` calling `logout` hooks could be an issue.

You are proposing the opposite though (`logout` triggering `removeLogin` hooks), which I think won't hurt and it also makes sense that logout removes login :)

Reply:
We are actually doing that in our application i.e. calling `removeLogin` hooks on `logout`.

Reply:
Done. Logout triggers `removeLogin` hooks as well.